src/components/config/WaypointVisibilityPanel.tsx

```diff
@@ -0,0 +1,75 @@
+import { observer } from "mobx-react";
+import React, { Component } from "react";
+import DocumentManagerContext from "../../document/DocumentManager";
+import styles from "./WaypointConfigPanel.module.css";
+import { Slider } from "@mui/material";
+
+type Props = {};
+
+type State = {};
+
+class WaypointVisibilityPanel extends Component<Props, State> {
+  static contextType = DocumentManagerContext;
+  declare context: React.ContextType<typeof DocumentManagerContext>;
+  state = {};
+  render() {
+    let startIndex =
+      this.context.model.document.pathlist.activePath
+        .visibleWaypointsRange[0] ?? 0;
+    let endIndex =
+      this.context.model.document.pathlist.activePath
+        .visibleWaypointsRange[1] ?? 0;
+    let points = this.context.model.document.pathlist.activePath.waypoints;
+    let pointcount = points.length;
+
+    const sliderMarks = [
```

Comment:
You don't need to add the Start and End marks here, and in fact, they're breaking your current parsing in the `onChange` handler. You also don't need to be filtering out initial guess points. You could break this down to 
`const sliderMarks = points.map((point, idx) => ({value: idx, label: idx}));`